src/tasks/thread_pool.hpp

```diff
@@ -89,6 +91,37 @@ class ThreadQueue {
   bool waiting = false;
 };
 
+template <typename T>
+class ThreadVector {
```

Comment:
I guess this is just a thread safe vector? If so, maybe a short comment clarifying it would be helpful.